src/kernel/capinv.c

```diff
@@ -365,14 +368,33 @@ cap_cpy(struct captbl *t, capid_t cap_to, capid_t capin_to, capid_t cap_from, ca
 		if (unlikely(!ctto)) return -ENOENT;
 		if (unlikely(ctto->type != cap_type)) return -EINVAL;
 		if (unlikely(((struct cap_pgtbl *)ctto)->refcnt_flags & CAP_MEM_FROZEN_FLAG)) return -EINVAL;
-		f = pgtbl_lkup_pte(((struct cap_pgtbl *)ctfrom)->pgtbl, capin_from, &flags);
+
+		/* 
+		 * See what kind of delegation we are doing. There are 4 kinds of delegations:
+		 * 1. Superpage -> Smallpage [capin_from_pos < 1024]
+		 * 2. Superpage -> Superpage [capin_from_pos == 0]
+		 * 3. Smallpage -> Smallpage [capin_from_pos == 0]
```

Comment:
What is `capin_from_pos` referring to? Just `capin_from`?

Reply:
This already deleted.